cpp/autosar/src/rules/A0-1-2/UnusedReturnValue.ql

```diff
@@ -39,8 +40,16 @@ where
   // so the rule does not require the use of the return value
   not f instanceof Operator and
   // Exclude cases where the function call is generated within a macro, as the user of the macro is
-  // not necessarily able to address thoes results
+  // not necessarily able to address those results
   not fc.isAffectedByMacro() and
-  // Rule allows disabling this rule where a static_cast<void> is applied
-  not fc.getExplicitlyConverted().(StaticCast).getActualType() instanceof VoidType
+  // Rule allows disabling this rule where a static_cast<void> or a C-style cast to void is applied
+  not (
+    fc.getExplicitlyConverted().(StaticCast).getActualType() instanceof VoidType
+    or
+    exists(CStyleCast cast |
+      not cast.isCompilerGenerated() and
+      cast.getExpr() = fc and
+      cast.getActualType() instanceof VoidType
```

Comment:
This case should replicate the first case, e.g.
```
or
fc.getExplicitlyConverted().(CStyleCast).getActualType() instanceof VoidType
```
As I don't think there's any intended semantic difference between the two case, so they should be identical in structure.

Note: `getExplicitlyConverted()` is the right concept for this query, because we want casts that are explicit in the code. An implicit conversion is not sufficient as "sanitizer" for this query. `not cast.isCompilerGenerated() and cast.getExpr() = fc` mostly does what you want in terms of excluding implicit conversions, but has some strange behaviour in the case that there are both explicit and implicit conversions, or parenthesis (which are also treated as conversions). In these cases, the conversions "stack", so `cast.getExpr()` may return another conversion - for example, consider `(foo) (f())`.

`getExplicitlyConverted()` ensures we get the explicit conversion associated with the function call return value.

An alternative to repeating the cases would be to unify them:
```
exists(Cast c |
  c instanceof StaticCast
  or
  c instanceof CStyleCast
  |
  c= fc.getExplicitlyConverted() and
  c.getActualType() instanceof VoidType
)
 ```